File: javascript/largest-series-product/largest-series-product.js
export function largestProduct(input, len) {
    if (isNaN(+input) || len < 0) {
        throw new Error('Invalid input.');
    }

    if (len > input.length) {
        throw new Error('Slice size is too big.');
    }

    if (!input || len === 0) {
        return 1;
    }

    const products = Array(input.length - len + 1)
        .fill(null)
        .map((_, i) => input.slice(i, i + len))
        .map(n => [...n].map(Number).reduce((a, b) => a * b));

    return Math.max(...products);
}

File: python/largest-series-product/largest_series_product.py
from operator import mul
from functools import reduce

def largest_product(series, size):
    if size < 0:
        raise ValueError('Invalid size')

    chunks = (series[i:i+size] for i in range(len(series)-size+1))
    products = (reduce(mul, map(int, chunk), 1) for chunk in chunks)
    return max(products)
